arangod/MMFiles/MMFilesCompactorThread.cpp

```diff
@@ -905,18 +905,26 @@ void MMFilesCompactorThread::run() {
         } catch (...) {
           collections.clear();
         }
-
+  
         for (auto& collection : collections) {
           bool worked = false;
+            
+          if (engine->isCompactionDisabled()) {
```

Comment:
What happens if this switch is switched off or on in the middle of this loop? Then the queued operations are processed for some collections but not for others. Will this lead to trouble?

Reply:
the compactor thread does not queue anything. it will look at one collection at a time, and compact it if it qualifies for compaction. collections that do not qualify are just skipped.
so when turning off compaction in the middle of the loop, we go into the skipping mode for all collections. this should not cause any trouble.